src-electron/generator/helper-attribute.js

```diff
@@ -76,6 +79,89 @@ async function attributeDefault(options) {
   return templateUtil.templatePromise(this.global, p)
 }
 
+/**
+ * Given an attribute Id determine its corresponding atomic identifier from the
+ * atomic table.
+ * @param {*} attributeId
+ */
+async function as_underlying_atomic_identifier_for_attribute_id(attributeId) {
```

Comment:
I might be wrong but doesn't matter have derived types from integers and floats. What would happen in that case?

Reply:
Don't know of any derived floats, ints or strings. Could you point me to any you ran across them?
if that is the case then need to add additional logic to handle that.

Reply:
does selectAtomicType() take care of that?

Reply:
I don't think we have any other tags apart from enum, bitmap and struct so I doubt it.

Reply:
Okay I'm getting confused here, we don't NEED to return baseline atomic types here right?

Reply:
So returning a derived atomic data type like clusterId would be acceptable here?

Reply:
does selectAtomicType() take care of that? => Yes that takes care of strings, ints, floats

Reply:
This has nothing to do with base types. It is about returning the Atomic_identfier from the atomic table based on attribute type

Reply:
Got it, sorry for the confusion.